File: game-rest-api/src/it/klotski/web/game/domain/game/Game.java
package it.klotski.web.game.domain.game;

import it.klotski.web.game.domain.user.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;

/**
 * Questa classe rappresenta un gioco Klotski.
 * Un oggetto Game contiene informazioni sul gioco, come l'identificatore, il giocatore associato,
 * l'ID della configurazione di partenza, la durata del gioco, le informazioni sulla creazione e l'aggiornamento,
 * il numero di mosse effettuate e lo stato di completamento del gioco.
 */
@Setter
@Entity
@Getter
@Table(name = "games")
@NoArgsConstructor
@AllArgsConstructor
@Builder
@With
@EqualsAndHashCode
public class Game implements IGame {
    /**
     * Identificatore univoco del gioco.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    /**
     * Il giocatore associato al gioco.
     */
    @OneToOne
    private User player;

    /**
     * L'ID della configurazione di partenza del gioco.
     */
    private int startConfigurationId;

    /**
     * Data e ora di creazione del gioco.
     */
    @CreationTimestamp
    private Timestamp createdAt;

    /**
     * Data e ora di aggiornamento del gioco.
     */
    @UpdateTimestamp
    private Timestamp updatedAt;

    /**
     * Lo stato di completamento del gioco.
     */
    private boolean finished;
}

File: web-client/src/types/models/Game.ts
import Board from "@/types/models/Board.ts";

/**
 * @class Game
 * @classdesc Rappresenta una partita di gioco.
 */
export default class Game {
    /**
     * @member {number} id - ID della partita.
     */
    id: number;

    /**
     * @member {number} startConfigurationId - ID della configurazione di partenza.
     */
    startConfigurationId: number;

    /**
     * @member {string} date - Data della partita.
     */
    date: string;

    /**
     * @member {boolean} finished - Flag che indica se la partita è terminata.
     */
    finished: boolean;

    /**
     * @member {number} moves - Numero di mosse effettuate durante la partita.
     */
    moves: number;

    /**
     * @member {Board} board - La scacchiera di gioco.
     */
    board: Board;

    /**
     * @constructor
     * @param {number} id - ID della partita.
     * @param {number} startConfigurationId - ID della configurazione di partenza.
     * @param {string} date - Data della partita.
     * @param {boolean} finished - Flag che indica se la partita è terminata.
     * @param {number} moves - Numero di mosse effettuate durante la partita.
     * @param {Board} board - La scacchiera di gioco.
     */
    constructor(id: number, startConfigurationId: number, date: string, finished: boolean, moves: number, board: Board) {
        this.id = id;
        this.startConfigurationId = startConfigurationId;
        this.date = date;
        this.finished = finished;
        this.moves = moves;
        this.board = board;
    }
}